cmd/dp-recipe-api/main.go

```diff
@@ -36,87 +42,59 @@ func main() {
 
 	//Feature Flag for Mongo Connection
 	enableMongoData := cfg.MongoConfig.EnableMongoData
-
-	if enableMongoData {
-		mongodb := &mongo.Mongo{
-			Collection: cfg.MongoConfig.Collection,
-			Database:   cfg.MongoConfig.Database,
-			URI:        cfg.MongoConfig.BindAddr,
-		}
-
-		var err error
-		mongodb.Session, err = mongodb.Init()
-		if err != nil {
-			log.Event(ctx, "failed to initialise mongo", log.FATAL, log.Error(err))
-			os.Exit(1)
-		}
+  
+	mongodb := &mongo.Mongo{
+		Collection: cfg.MongoConfig.Collection,
+		Database:   cfg.MongoConfig.Database,
+		URI:        cfg.MongoConfig.BindAddr,
 	}
-
-	router := mux.NewRouter()
-	router.Methods("GET").Path("/health").HandlerFunc(healthcheck)
-	router.Methods("GET").Path("/recipes").HandlerFunc(recipeListHandler)
-	router.Methods("GET").Path("/recipes/{id}").HandlerFunc(recipeHandler)
-
-	log.Event(ctx, "starting http server", log.INFO, log.Data{"bind_addr": cfg.BindAddr})
-	srv := server.New(cfg.BindAddr, router)
-	if err := srv.ListenAndServe(); err != nil {
-		log.Event(ctx, "error starting http server for API", log.FATAL, log.Error(err))
+  
+	var err error
+	mongodb.Session, err = mongodb.Init()
```

Comment:
Should we be initialising Mongo if the feature flag is not enabled?

Reply:
Initially that's what I did but then it became a problem when the gracefulShutdown was implemented. At first, the initialisation of `mongodb`was created within in the `if enableMongoData`. This means that `mongodb` would be only initialised when enabled.

But when implementing the gracefulShowdown, i cannot call the variable `mongodb` as i would get the error of mongodb being undefined in gracefulShowdown function. This is due to the fact that it is initialised in an if statement. 

That's why I coded in such a way that the initialised `mongodb`is only used when creating the RecipeAPI instance with Mongo in datastore when enabled

Is there another way in which I can achieve it?

Reply:
It should be fine to move the initialisation into the `if enableMongoData`block. If the reference to `mongodb` still lives outside of the if statement, and you also have the if statement in the graceful shutdown (which you already do). Give it a try and let me know if you have any issues 👍

Reply:
Done in commit "Initialising mongo session within feature flag"